pyteal/compiler/compiler_test.py

```diff
@@ -2195,2122 +2199,264 @@ def access_b4_store(magic_num: pt.abi.Uint64, *, output: pt.abi.Uint64):
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-frame_dig 1
-frame_dig 2
-callsub add_0
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// sub_caster
-subcaster_10:
-proto 0 0
-int 0
-dupn 2
-txna ApplicationArgs 1
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-frame_dig 1
-frame_dig 2
-callsub sub_1
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// mul_caster
-mulcaster_11:
-proto 0 0
-int 0
-dupn 2
-txna ApplicationArgs 1
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-frame_dig 1
-frame_dig 2
-callsub mul_2
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// div_caster
-divcaster_12:
-proto 0 0
-int 0
-dupn 2
-txna ApplicationArgs 1
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-frame_dig 1
-frame_dig 2
-callsub div_3
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// mod_caster
-modcaster_13:
-proto 0 0
-int 0
-dupn 2
-txna ApplicationArgs 1
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-frame_dig 1
-frame_dig 2
-callsub mod_4
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// all_laid_to_args_caster
-alllaidtoargscaster_14:
-proto 0 0
-int 0
-dupn 16
-byte ""
-txna ApplicationArgs 1
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-txna ApplicationArgs 3
-btoi
-frame_bury 3
-txna ApplicationArgs 4
-btoi
-frame_bury 4
-txna ApplicationArgs 5
-btoi
-frame_bury 5
-txna ApplicationArgs 6
-btoi
-frame_bury 6
-txna ApplicationArgs 7
-btoi
-frame_bury 7
-txna ApplicationArgs 8
-btoi
-frame_bury 8
-txna ApplicationArgs 9
-btoi
-frame_bury 9
-txna ApplicationArgs 10
-btoi
-frame_bury 10
-txna ApplicationArgs 11
-btoi
-frame_bury 11
-txna ApplicationArgs 12
-btoi
-frame_bury 12
-txna ApplicationArgs 13
-btoi
-frame_bury 13
-txna ApplicationArgs 14
-btoi
-frame_bury 14
-txna ApplicationArgs 15
-frame_bury 17
-frame_dig 17
-int 0
-extract_uint64
-frame_bury 15
-frame_dig 17
-int 8
-extract_uint64
-frame_bury 16
-frame_dig 1
-frame_dig 2
-frame_dig 3
-frame_dig 4
-frame_dig 5
-frame_dig 6
-frame_dig 7
-frame_dig 8
-frame_dig 9
-frame_dig 10
-frame_dig 11
-frame_dig 12
-frame_dig 13
-frame_dig 14
-frame_dig 15
-frame_dig 16
-callsub alllaidtoargs_5
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
+    router_compilarison(
+        _router_without_oc,
+        8,
+        "yaccFP_approval_v8.teal",
+        "yaccFP_clear_v8.teal",
+        "yacc",
+    )
 
-// empty_return_subroutine_caster
-emptyreturnsubroutinecaster_15:
-proto 0 0
-callsub emptyreturnsubroutine_6
-retsub
+    non_trivial = pt.Seq(
+        pt.Cond(
+            [pt.Txn.application_args.length() < pt.Int(2), pt.Approve()],
+            [pt.Txn.application_args[0] != pt.Bytes("CLEANUP"), pt.Approve()],
+            [pt.Txn.application_args[1] != pt.Bytes("ABORTING"), pt.Approve()],
+        ),
+        pt.Reject(),
+    )
```

Comment:
a bit refactoring: reject cannot be reached, if none of 3 conditions in cond are satisfied. By cond construction, it will error and break herre. So move reject into last cond branch, with condition being 1 for any other case to fall in.

Reply:
cf [87f281e](https://github.com/algorand/pyteal/pull/634/commits/87f281eecee511bb71588e821273cb8d2b0287e9)

Reply:
lookin good, resolving